iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/DanglingLineImpl.java

```diff
@@ -19,6 +17,166 @@
  */
 class DanglingLineImpl extends AbstractConnectable<DanglingLine> implements DanglingLine, CurrentLimitsOwner<Void> {
 
+    public static class GenerationImpl implements Generation, ReactiveLimitsOwner, Validable {
+
+        private final DanglingLineImpl danglingLine;
+
+        private final ReactiveLimitsHolderImpl reactiveLimits;
+
+        private double minP;
+
+        private double maxP;
+
+        // attributes depending on the variant
+
+        private final TDoubleArrayList targetP;
+
+        private final TDoubleArrayList targetQ;
+
+        private final TBooleanArrayList voltageRegulationOn;
+
+        private final TDoubleArrayList targetV;
+
+        GenerationImpl(DanglingLineAdderImpl.GenerationAdderImpl adder, DanglingLineImpl danglingLine) {
```

Comment:
This is not a good design: you should keep only the second constructor. Remove the DanglingLineImpl and male the generation class not static to be able to access to the DanglingLine.

Reply:
See my remark on the comment above